node/system_function.go

```diff
@@ -0,0 +1,71 @@
+package node
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/buildpack/libbuildpack/layers"
+)
+
+type SystemFunction struct {
+	Path  string `toml:"path"`
+	Layer layers.Layer
+}
+
+func NewSystemFunction(l layers.Layer) (SystemFunction, error) {
+	// TODO push this up into buildpack.go
+	buildpackDir, err := filepath.Abs(filepath.Dir(os.Args[0]))
+	if err != nil {
+		return SystemFunction{}, err
+	}
+
+	return SystemFunction{
+		Path:  filepath.Join(buildpackDir, "../system/"),
+		Layer: l,
+	}, nil
+}
+
+func (f SystemFunction) Contribute() error {
+	if err := f.Layer.WriteMetadata(f, layers.Launch); err != nil {
+		return err
+	}
+
+	if err := os.MkdirAll(f.Layer.Root, 0755); err != nil {
```

Comment:
Can @jkutner or @hone comment on this a bit? I wasn't really sure what part of this process should be creating the "system" folder in our case.

Reply:
not sure I understand, but this looks about right. Similar to https://github.com/heroku/java-buildpack/blob/master/maven/maven.go#L171